From 44ff1871c429e940e65d52a7d0f268b51741b851 Mon Sep 17 00:00:00 2001
From: Markus Kappeler <markus.kappeler@bytesatwork.ch>
Date: Mon, 1 May 2023 19:15:12 +0200
Subject: [PATCH] drivers: watchdog: rti_wdt: set min_hw_heartbeat_ms to 51%

When watchdog is triggered within the 50% then the system reboots.
So setting the min_hw_heartbeat_ms property to 51% to relax it.

Signed-off-by: Markus Kappeler <markus.kappeler@bytesatwork.ch>
---
 drivers/watchdog/rti_wdt.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/watchdog/rti_wdt.c b/drivers/watchdog/rti_wdt.c
index f91a260beaa7..21b1071cce74 100644
--- a/drivers/watchdog/rti_wdt.c
+++ b/drivers/watchdog/rti_wdt.c
@@ -87,8 +87,10 @@ static int rti_wdt_start(struct watchdog_device *wdd)
 	 * The HW configuration options only allow for the open window size
 	 * to be 50% or less than that; we obviouly want to configure the open
 	 * window as large as possible so we select the 50% option.
+	 * Set some margin for the min_hw_heartbeat_ms to 51% to be in no case
+	 * to early.
 	 */
-	wdd->min_hw_heartbeat_ms = 500 * wdd->timeout;
+	wdd->min_hw_heartbeat_ms = 510 * wdd->timeout;
 
 	/* Generate NMI when wdt expires */
 	writel_relaxed(RTIWWDRX_NMI, wdt->base + RTIWWDRXCTRL);
-- 
2.39.2

